packages/server/src/api/controllers/row/ExternalRequest.ts

```diff
@@ -248,6 +163,117 @@ export class ExternalRequest<T extends Operation> {
     if (datasource && datasource.entities) {
       this.tables = datasource.entities
     }
+    this.tableList = Object.values(this.tables)
+  }
+
+  private prepareFilters(
+    id: string | undefined | string[],
+    filters: SearchFilters,
+    table: Table
+  ) {
+    const tables = this.tableList
+    // replace any relationship columns initially, table names and relationship column names are acceptable
+    const relationshipColumns = sdk.rows.filters.getRelationshipColumns(table)
+    filters = sdk.rows.filters.updateFilterKeys(
+      filters,
+      relationshipColumns
+        .map(({ name, definition }) => {
+          const { tableName } = breakExternalTableId(definition.tableId)
+          return {
+            original: name,
+            updated: tableName!,
```

Comment:
It's a bit annoying that this is necessary. It seems like `breakExternalTableId` is one of those functions that could take nothing and if it does it returns nothing. And I can see the value in that leniency, but it does create situations like this where you have to override the type checker. I wonder if it would be difficult to make `breakExternalTableId` only take known strings rather than undefined or to type it in such a way that if it doesn't receive undefined it doesn't return an undefined result.

Reply:
Just went through and updated this - it wasn't actually too bad!